src/algorithms/preparePlotData.ts

```diff
@@ -15,6 +15,7 @@ export function preparePlotData(trajectory: Trajectory): PlotDatum[] {
       overflow: verifyPositive(x.current.overflow.total),
       recovered: verifyPositive(x.cumulative.recovered.total),
       fatality: verifyPositive(x.cumulative.fatality.total),
+      weeklyFatality: verifyPositive(x.cumulative.fatality.total - middle[i > 6 ? i - 7 : 0].cumulative.fatality.total),
```

Comment:
@rneher 
If I read it correctly, the conditionals `i > 6 ? i - 7 : 0`, or even the entire subtraction expressions seem to be duplicated. Could we extract them into a variable and bubble it up ? For the sake of self-documentation, some good name for this variable would be nice to have as well. I'm not sure if I correctly understand what this expression calculates and why.

This refactoring will probably require introducing a function scope with `return` statement and all this code will be indented to the right, so we may do this refactoring afterwards, to keep the current diff clear, as it is.

Reply:
we should call it `day`! and yes, its duplicated. But is hard to see how the function call is any shorter than the arguments. Maybe adding a comment is good enough...

Reply:
I never said "shorter"! All I care is the programmer's performance (code that is easy to understand) and runtime performance (avoiding unnecessary computation).

Not function call, just a variable:

```js
something.map((...) => {
   const day = i > 6 ? i - 7 : 0
   const thingy = something - middle[day].cumulative.fatality.total

   return {
     one: verify(a - thingy),
     two: verify(b - thingy),
     three: verify(c - thingy),
  }
}
```

or better (so that the usage of the variable is closer to the declaration):

```js
something.map((...) => {
   const day = i > 6 ? i - 7 : 0
   const thingy = middle[day].cumulative.fatality.total

   const one = verify(a - thingy)
   const two = verify(b - thingy)
   const three = verify(c - thingy)

   return { one, two, three }
}
```

also maybe

```js

  const one = a - thingy
  const two = b - thingy
  const three = c - thingy

  return ensureSomeParticularPlotDataRequirements({ one, two, three })

```
to cut the duplication further.

Anyways, refactoring is out of scope of this PR I think. The clear diff is more important here, to see the changes. We can sit down and refactor later.

Reply:
I vaguely remember:
`#define SEVEN_DAYS_AGO(d) ( d>6 ? d-7 : 0)`